This failure seems unrelated, it also happened in https://github.com/zephyrproject-rtos/zephyr/pull/109272#issuecomment-4635762703

https://github.com/zephyrproject-rtos/zephyr/actions/runs/27041307945/job/79817712641
```
        b.run_cmake = True
        b.build_dir = str(tmp_path / 'build')
    
>       b._run_cmake(board='native_sim', origin=None)

tests/test_build.py:229: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
build.py:694: in _run_cmake
    user_args = self.config.get('build.cmake-args', default=None)
                ^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/west/commands.py:293: in _get_config
    self.die(f"can't run west {self.name}; it requires config "
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/west/commands.py:519: in die
    self.err(*args, fatal=True)
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/west/commands.py:502: in err
    if self.color_ui:
       ^^^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/west/commands.py:529: in color_ui
    return self.config.getboolean('color.ui', default=True)
           ^^^^^^^^^^^
/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/site-packages/west/commands.py:293: in _get_config
    self.die(f"can't run west {self.name}; it requires config "
E   RecursionError: maximum recursion depth exceeded
!!! Recursion detected (same locals & position)
=========================== short test summary info ============================
FAILED tests/test_build.py::test_test_item_updates_source_dir - RecursionError: maximum recursion depth exceeded
=================== 1 failed, 882 passed, 2 skipped in 2.77s ===================
Traceback (most recent call last):
  File "/Users/runner/work/zephyr/zephyr/./scripts/west_commands/run_tests.py", line 37, in <module>
    subprocess.run(command, check=True, cwd=here)
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/Library/Frameworks/Python.framework/Versions/3.13/bin/python3', '-m', 'pytest']' returned non-zero exit status 1.
Error: Process completed with exit code 1.

```

EDIT: same again in https://github.com/zephyrproject-rtos/zephyr/actions/runs/27066031213/job/79886940654?pr=110671

Hotfix submitted.

--- a/scripts/west_commands/build_helpers.py
+++ b/scripts/west_commands/build_helpers.py
@@ -23,8 +23,7 @@ import zcmake
 
 # Explicit, flat name: avoids colliding with scripts/pylib/build_helpers/
 # domains.py (which grabs `logging.getLogger('build_helpers')` at import
-# time), and avoids the `west.*` namespace whose NullHandler would defeat
-# the hasHandlers() guard in forward_logging_to_west.
+# time).
 BUILD_HELPERS_LOGGER = 'zephyr_build_helpers'
 _logger = logging.getLogger(BUILD_HELPERS_LOGGER)
 
@@ -79,18 +78,51 @@ def forward_logging_to_west(command, logger_names):
 
     ``logger_names`` may be a single logger name or an iterable of names.
 
-    Safe to call more than once: handlers are not duplicated.
+    This simply adds a custom handler to each logger. If you forget to
+    forward to any West command, then Python's ``logging.lastResort``
+    should still print warnings and higher to stderr while discarding
+    lower priorities, see the
+    https://docs.python.org/3/library/logging.html HOWTO for details.
+
+    While not tested, you could in theory forward to different West
+    commands at the same time. Forwarding multiple times to the _same_
+    West command will print an error and be ignored.
     '''
     if isinstance(logger_names, str):
         logger_names = [logger_names]
     for name in logger_names:
         logger = logging.getLogger(name)
-        # Drop DEBUG records at the logger unless the user asked for them,
-        # so they don't reach attached handlers (ours or anyone else's).
-        logger.setLevel(1 if command.verbosity >= Verbosity.DBG else logging.INFO)
-        if not logger.hasHandlers():
-            # Only add a log handler if none has been added already.
-            logger.addHandler(WestLogHandler(command))
+        _fwd_logger(logger, command)
+
+
+def _fwd_logger(logger, west_cmd):
+
+    # Note logger.handlers are only the _directly_ attached handlers,
+    # ignoring propagation up
+    for handler in logger.handlers:
+        match handler:
+            case WestLogHandler():
+                prev_cmd = handler._command
+                hdlr_print = f"WestLogHandler({prev_cmd.name})"
+                if prev_cmd == west_cmd:
+                    west_cmd.err(
+                        f"logger('{logger.name}') was already forwarded to {hdlr_print}!"
+                    )
+                    return
+            case _:
+                hdlr_print = f"{handler}"
+
+        west_cmd.dbg(f"logger('{logger.name}') already had handler: {hdlr_print}")
+
+    # Drop DEBUG records at the logger unless the user asked for them,
+    # so they don't reach attached handlers (ours or anyone else's).
+    logger.setLevel(1 if west_cmd.verbosity >= Verbosity.DBG else logging.INFO)
+
+    logger.addHandler(WestLogHandler(west_cmd))
+    # Use immediately.  This starts with "logger %(name)s" thanks
+    # to WestLogHandler.__init__() above
+    logger.debug(f"forwarded to WestLogHandler({west_cmd.name}). propagate={logger.propagate}")
+
 
 # Domains.py must be imported from the pylib directory, since
 # twister also uses the implementation

--- a/scripts/west_commands/run_common.py
+++ b/scripts/west_commands/run_common.py
@@ -313,6 +313,9 @@ def do_run_common(command, user_args, user_runner_args, domain_file=None):
         if len(entry.boards) == 0:
             del used_cmds[i]
 
+    # Set up runner logging to delegate to the WestCommand logging methods.
+    forward_logging_to_west(command, 'runners')
+
     prev_runner = None
     for d in domains:
         prev_runner = do_run_common_image(command, user_args, user_runner_args, used_cmds,
@@ -341,9 +344,6 @@ def do_run_common_image(command, user_args, user_runner_args, used_cmds,
     runner_cls = use_runner_cls(command, board, user_args, runners_yaml,
                                 cache)
     runner_name = runner_cls.name()
-
-    # Set up runner logging to delegate to the WestCommand logging methods.
-    forward_logging_to_west(command, 'runners')
 
     # If the user passed -- to force the parent argument parser to stop
     # parsing, it will show up here, and needs to be filtered out.

